lib/cpp/src/thrift/protocol/TJSONProtocol.cpp

```diff
@@ -736,7 +741,23 @@ uint32_t TJSONProtocol::readJSONString(std::string& str, bool skipContext) {
       ch = reader_.read();
       ++result;
       if (ch == kJSONEscapeChar) {
-        result += readJSONEscapeChar(&ch);
+        uint16_t cp;
+        result += readJSONEscapeChar(&cp);
+        // Checking for surrogate pair
+        if (cp >= 0xD800 && cp <= 0xDBFF) {
+          codepoints.push_back(cp);
+        } else {
+          if (cp >= 0xDC00 && cp <= 0xDFFF
```

Comment:
Does 0xDC00 and 0xDFFF have some names? it now looks like some magic number to me :) (btw - instead of adding comment I'd extract it as function, named like comment)